replica/backup.go

```diff
@@ -175,24 +180,92 @@ func (rb *Backup) findIndex(id string) int {
 	return -1
 }
 
+type Hole struct {
+	f      types.DiffDisk
+	offset int64
+	len    int64
+}
+
+var HoleCreatorChan = make(chan Hole, 1000000)
+
+//CreateHoles removes the offsets from corresponding sparse files
+func CreateHoles() error {
+	retryCount := 0
+	for {
+		hole := <-HoleCreatorChan
+	retry:
+		if err := syscall.Fallocate(int(hole.f.Fd()),
```

Comment:
When a diffDisk is used here, it should not get deleted from other ways like 'snap destroy' etc

Reply:
This will be covered in snap destroy PR.

Reply:
as we don't delete internal snapshots, it should be fine.